core/ibc/src/handler.rs

```diff
@@ -0,0 +1,30 @@
+use std::sync::Arc;
+
+use once_cell::sync::OnceCell;
+
+use protocol::traits::IbcAdapter;
+
+pub static IBC_HANDLER: OnceCell<Box<dyn IbcHandle>> = OnceCell::new();
+
+pub trait IbcHandle: Send + Sync {
+    fn handle(&self, raw: &[u8]) -> u8;
+}
+
+pub struct IbcHandlerImpl<Adapter> {
+    _adapter: Arc<Adapter>,
+}
+
+impl<Adapter: IbcAdapter + 'static> IbcHandle for IbcHandlerImpl<Adapter> {
+    fn handle(&self, _raw: &[u8]) -> u8 {
```

Comment:
The axon implimentation of the `deliver` https://github.com/informalsystems/basecoin-rs/blob/main/src/app/modules/ibc.rs#L297 of `basecoin-rs` ?